documentation/library_guide/parallel_api/pass_data_algorithms.rst

```diff
@@ -85,13 +85,17 @@ Alternatively, use ``std::vector`` with a USM allocator.
 
 Note: The ability to appropriately detect USM allocated ``std::vector::iterator`` depends
 on details of the C++ standard library implementation and what information about the
-allocator is included in the ``std::vector::iterator`` type definition. If USM allocated
-vector iterators are not detectable with your C++ standard library, they will still function
-as inputs to oneDPL, but they will be treated as if they were host-side
-``std::vector::iterator`` as described in the section below. To guarantee no additional
-host-side copies, you can use ``std::vector::data()`` in combination with ``std::vector::size()``
-with USM allocated vectors to obtain a USM pointers. This will avoid extra host-side copies
-regardless of the C++ standard library implementation.
+allocator is included in the ``std::vector::iterator`` type definition. You can check
+if your C++ standard library implementation provides enough information by checking the
+value of ``oneapi::dpl::usm_allocated_vector_iterators_supported_v<ValueType>``. This will
+evaluate to ``true`` for C++ standard library implementations which allow oneDPL to support
```

Comment:
Please add a comma between "implementations" and "which" and add an "s" to "allow". The final sentence should read: 

This will evaluate to ``true`` for C++ standard library implementations, which allows oneDPL to support USM allocators with this ValueType and ``false`` otherwise.

Reply:
I think there was some confusion by my wording.  I've expanded this some to hopefully  make things more clear.

Reply:
Thanks for the update. My suggestions on the new content (ignore spacing; I broke things up for this review). 

This evaluates to ``true`` when using a C++ standard library implementation, which allows oneDPL to detect and
support USM allocators for ``ValueType`` and ``false``. 

When ``oneapi::dpl::usm_allocated_vector_iterators_supported_v<ValueType>`` evaluates to ``false``,
you should rely on USM pointers directly instead of using iterators to vectors with USM allocators.